node/overseer/overseer-gen/proc-macro/src/impl_builder.rs

```diff
@@ -107,46 +62,243 @@ pub(crate) fn impl_builder(info: &OverseerInfo) -> proc_macro2::TokenStream {
 		})
 		.collect::<Vec<_>>();
 
-	let generics = quote! {
-		< S, #( #baggage_generic_ty, )* >
-	};
-	let where_clause = quote! {
+	// Helpers to use within quote! macros
+	let subsystem_where_clause = &info
+		.subsystems()
+		.iter()
+		.map(|subsys| {
+			let generic_ty = &subsys.generic;
+			let consumes = &subsys.consumes;
+			quote! {
+				#generic_ty : #support_crate ::Subsystem<#subsystem_ctx_name< #consumes >, #error_ty>
+			}
```

Comment:
could be a `WherePredicate` collected into a `Punctuated<WherePredicate, Comma>,`